src/Sylius/Bundle/ApiBundle/CommandHandler/Cart/PickupCartHandler.php

```diff
@@ -72,10 +74,7 @@ public function __invoke(PickupCart $pickupCart)
         $cart->setCurrencyCode($currency->getCode());
         $cart->setTokenValue($pickupCart->tokenValue ?? $this->generator->generateUriSafeString(10));
 
-        if ($customer !== null) {
-            $cart->setCustomerWithAuthorization($customer);
-            $cart->setBillingAddress($this->getDefaultAddress($customer));
-        }
+        $this->checkIfCustomerIsAGueast($customer, $cart);
```

Comment:
No point in extracting the code like this, also the method name is wrong, since it does other stuff than just checking if its a guest or not.